### PR TITLE
Add assessments to Temporary Accommodation applications

### PR DESCRIPTION
**Pull request by danlivings-dxw**
> See [ticket #1287 on the CAS3 Trello board](https://trello.com/c/72apWgsr/1287-the-frontend-can-get-applications-for-an-hpt).

This PR enables assessments to be used for the Temporary Accommodation service. Assessments for Temporary Accommodation have differences to those for Approved Premises, namely:
- They are not automatically allocated to an assessor, and must be marked as explicitly being in progress.
- The list of assessments visible to a user can be sorted in various ways, or filtered by the status.
- Temporary Accommodation assessments require a different set of statuses from Approved Premises assessments, but this has not been implemented within this PR (it will be implemented as part of the upcoming [ticket #1308](https://trello.com/c/X2ymPmWq/1308-application-state-can-be-updated-by-the-frontend)).

### CODE REVIEW

**Review by brownbearsoftware — APPROVED** [review]
A lot going on here but can't see any problems

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/AssessmentService.kt [hunk 1]
```diff
@@ -107,19 +110,24 @@ class AssessmentService(
...
   }
 
-  fun createAssessment(application: ApplicationEntity): AssessmentEntity {
-    if (application !is ApprovedPremisesApplicationEntity) {
-      throw RuntimeException("Only CAS1 Applications are currently supported")
-    }
+  fun createAssessment(application: ApplicationEntity): AssessmentEntity = when (application) {
+    is ApprovedPremisesApplicationEntity -> createApprovedPremisesAssessment(application)
+    is TemporaryAccommodationApplicationEntity -> createTemporaryAccommodationAssessment(application)
+    else -> throw RuntimeException("Application type '${application::class.simpleName}' is not supported")
```
Comment (danlivings-dxw) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/AssessmentService.kt:116]: Should change to `qualifiedName` to be consistent with elsewhere

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/AssessmentTest.kt [hunk 2]
```diff
@@ -111,6 +118,75 @@ class AssessmentTest : IntegrationTestBase() {
...
+
+          AssessmentParams(assessment, offenderDetails, inmateDetails)
+        }
+
+        val expectedAssessments = when (sortField) {
+          AssessmentSortField.personName -> assessments.sortedByDescending { "${it.offenderDetails.firstName} ${it.offenderDetails.surname}" }
+          AssessmentSortField.personCrn -> assessments.sortedByDescending { it.assessment.application.crn }
+          AssessmentSortField.assessmentArrivalDate -> assessments.sortedByDescending { (it.assessment.application as TemporaryAccommodationApplicationEntity).arrivalDate }
+          AssessmentSortField.assessmentStatus -> {
+            Assumptions.assumeThat(true).isFalse
```
Comment (danlivings-dxw) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/AssessmentTest.kt:236]: Comment explaining this has disappeared, reintroduce

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/util/AssessmentUtils.kt [hunk 3]
```diff
@@ -61,3 +65,13 @@ private fun List<AssessmentSummary>.sort(sortOrder: SortOrder?, sortField: Asses
 
   return this
 }
+
+private fun List<AssessmentSummary>.filterByStatuses(statuses: List<AssessmentStatus>?): List<AssessmentSummary> {
+  if (statuses != null) {
+    val statusesSet = statuses.toSet()
+    println(statusesSet)
+    return this.filter { println(it.status); statusesSet.contains(it.status) }
```
Comment (danlivings-dxw) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/util/AssessmentUtils.kt:73]: Remove `println` statement

**Pull request merged**